cli/tsc/99_main_compiler.js

```diff
@@ -862,12 +862,25 @@ delete Object.prototype.__proto__;
     }
   }
 
+  /** @param {Record<string, string>} config */
+  function normalizeConfig(config) {
+    // the typescript compiler doesn't know about the precompile
+    // transform at the moment, so just tell it we're using react-jsx
+    if (config.jsx === "precompile") {
+      config.jsx = "react-jsx";
+    }
+    return config;
```

Comment:
I added this code... not sure it's ok temporarily for type checking purposes... probably not 😅